internals/overlord/logstate/gatherer.go

```diff
@@ -288,30 +301,22 @@ func (t *timer) EnsureSet(timeout time.Duration) {
 // protocol required by that log target.
 // For example, a logClient for Loki would encode the log messages in the
 // JSON format expected by Loki, and send them over HTTP(S).
-//
-// logClient implementations have some freedom about the semantics of these
-// methods. For a buffering client (e.g. HTTP):
-//   - Write could add the log to the client's internal buffer, calling Flush
-//     when this buffer reaches capacity.
-//   - Flush would prepare and send a request with the buffered logs.
-//
-// For a non-buffering client (e.g. TCP), Write could serialise the log
-// directly to the open connection, while Flush would be a no-op.
 type logClient interface {
-	// Write adds the given log entry to the client. Depending on the
-	// implementation of the client, this may send the log to the remote target,
-	// or simply add the log to an internal buffer, flushing that buffer when
-	// required.
-	Write(context.Context, servicelog.Entry) error
-
-	// Flush sends buffered logs (if any) to the remote target. For clients which
-	// don't buffer logs, Flush should be a no-op.
+	// AddLog adds the given log entry to the client's buffer.
+	AddLog(servicelog.Entry) error
```

Comment:
I think Gustavo's point was that we should still buffer in the case of syslog - so we're not making a syscall for every single log message. Otherwise, we can just set the buffer size to 1 for syslog and it's effectively the same behaviour. Either way, no context needed for `Add`.

Reply:
and even if syslog does send directly to the stream, this is a non-blocking / async operation (we're not waiting for a response from anything). So I maintain the context is unnecessary.